yb-voyager/src/dbzm/yb_cdc_client.go

```diff
@@ -0,0 +1,176 @@
+	}
+	ybc.ybCdcClientJarPath = fmt.Sprintf("%s/yb-client-cdc-stream-wrapper.jar", DEBEZIUM_DIST_DIR)
+	return nil
+}
+
+func (ybc *YugabyteDBCDCClient) GetStreamID() (string, error) {
+	streamID, err := ybc.readYBStreamID()
+	if err != nil && strings.Contains(err.Error(), "stream id not found") {
+		streamID, err = ybc.GenerateAndStoreStreamID()
+		if err != nil {
+			return "", fmt.Errorf("failed to generate and store stream id: %w", err)
+		}
+		utils.PrintAndLog("Generated YugabyteDB CDC stream-id: %s", streamID)
+		return streamID, nil
+	} else if err != nil {
+		return "", fmt.Errorf("failed to read stream id: %w", err)
+	}
+	utils.PrintAndLog("Using YugabyteDB CDC stream-id: %s", streamID)
+	return streamID, nil
+}
+
+func (ybc *YugabyteDBCDCClient) GenerateAndStoreStreamID() (string, error) {
+	args := fmt.Sprintf("-create -master_addresses %s -table_name %s -db_name %s ", ybc.masterAddreses, ybc.tableName, ybc.dbName)
+
+	if ybc.sslRootCert != "" {
+		args += fmt.Sprintf(" -ssl_cert_file %s", ybc.sslRootCert)
+	}
+
+	stdout, err := ybc.runCommand(args)
+	if err != nil {
+		return "", fmt.Errorf("running command with args: %s, error: %s", args, err)
+	}
+	//stdout - CDC Stream ID: <stream_id>
+	streamID := strings.Trim(strings.Split(stdout, ":")[1], " \n")
+	//storing streamID in a file
+	streamIDFile := filepath.Join(ybc.exportDir, "metainfo", "yb_cdc_stream_id.txt")
+	file, err := os.Create(streamIDFile)
+	if err != nil {
+		return "", fmt.Errorf("creating file: %s, error: %s", streamIDFile, err)
+	}
+	defer file.Close()
+	_, err = file.WriteString(streamID)
+	if err != nil {
+		return "", fmt.Errorf("writing to file: %s, error: %s", streamIDFile, err)
+	}
+	return streamID, nil
+}
+
+func (ybc *YugabyteDBCDCClient) readYBStreamID() (string, error) {
+	streamIDFilePath := filepath.Join(ybc.exportDir, "metainfo", "yb_cdc_stream_id.txt")
+	if utils.FileOrFolderExists(streamIDFilePath) {
+		streamID, err := os.ReadFile(streamIDFilePath)
+		if err != nil {
+			return "", fmt.Errorf("failed to read stream id file: %w", err)
+		}
+		return string(streamID), nil
+	}
+	return "", fmt.Errorf("yugabytedb cdc stream id not found at %s", streamIDFilePath)
+}
+
+func (ybc *YugabyteDBCDCClient) DeleteStreamID() error {
+	streamID, err := ybc.readYBStreamID()
+	if err != nil {
+		return fmt.Errorf("failed to read stream id: %w", err)
+	}
+	args := fmt.Sprintf("-delete_stream %s -master_addresses %s ", streamID, ybc.masterAddreses)
+
+	if ybc.sslRootCert != "" {
+		args += fmt.Sprintf(" -ssl_cert_file %s", ybc.sslRootCert)
+	}
+
+	_, err = ybc.runCommand(args)
+	if err != nil {
+		return fmt.Errorf("running command with args: %s, error: %s", args, err)
+	}
+	utils.PrintAndLog("Deleted YugabyteDB CDC stream-id: %s", streamID)
+	return nil
+}
+
+func (ybc *YugabyteDBCDCClient) ListMastersNodes() (string, error) {
+	args := fmt.Sprintf("-list_masters -master_addresses %s ", ybc.masterAddreses)
```

Comment:
Is it okay to pass both master and non-master nodes to the `-master_addresses` argument?

Reply:
There is no problem as such in this func as we are just listing the master nodes and it is working fine I tried so seems to be no problem there, but I think that to be on the safer side I should pass only master nodes to create and delete APIs after listing them so that there can never be a problem in future with those two APIs as they are crucial for streaming. Makes sense right?